r2/r2/controllers/api.py

```diff
@@ -1709,7 +1709,9 @@ def POST_block(self, thing):
             sr = None
 
         if getattr(thing, "from_sr", False) and sr:
-            BlockedSubredditsByAccount.block(c.user, sr)
+            # Users may only block a subreddit they don't mod
+            if not (thing.user_is_moderator or c.user_is_admin):
```

Comment:
The wrapped properties won't be available here (`user_is_moderator`) since it's a Thing being returned in the validator, so it will have to be `sr.is_moderator(c.user)`